core/services/workflows/delegate.go

```diff
@@ -54,11 +56,18 @@ func (d *Delegate) ServicesForSpec(ctx context.Context, spec job.Job) ([]job.Ser
 		go mercuryEventLoop(trigger, d.logger)
 	}
 
+	delayedExecution, err := initializeDelayedExecution(d.peerID)
+	if err != nil {
+		d.logger.Errorw("could not initialize delayed execution", err)
+		return nil, nil
+	}
+
 	cfg := Config{
-		Lggr:       d.logger,
-		Spec:       spec.WorkflowSpec.Workflow,
-		WorkflowID: spec.WorkflowSpec.WorkflowID,
-		Registry:   d.registry,
+		Lggr:                    d.logger,
+		Spec:                    spec.WorkflowSpec.Workflow,
+		WorkflowID:              spec.WorkflowSpec.WorkflowID,
+		Registry:                d.registry,
+		TargetExecutionStrategy: delayedExecution,
```

Comment:
Why is ExecutionStrategy part of Engine config and not Target config? It seems that a Strategy could be attached to every target (already configured on creation) and then Engine could call them individually. This will also make more sense for remote targets as their DON config can be different. WDYT?

Reply:
Yes, good idea 👍

Reply:
Done :) Let me know what you think!